sources/editor/Stride.Core.Assets.Editor/Services/UserDocumentationService.cs

```diff
@@ -89,65 +93,161 @@ private bool CacheAssemblyDocumentation([NotNull] Assembly assembly)
+            {
+                return;
+            }
+
+
+            var basePath = Path.Combine(Path.GetDirectoryName(location) ?? "", Path.GetFileNameWithoutExtension(location));
+            if (!CacheCustomDocumentation(basePath + ".usrdoc"))
+            {
+                Log.Info($"Could not cache from {basePath + ".usrdoc"}. Attempting to read from {basePath + ".xml"}");
+                // Fallback to XML assembly.
+                if (!CacheXmlDocumentation(basePath + ".xml"))
                 {
-                    undocumentedAssemblies.Add(assemblyName);
-                    return false;
+                    lock (lockObj)
+                    {
+                        undocumentedAssemblies.Add(assemblyName);
+                    }
+                    return;
                 }
-                try
+            }
+
+            lock (lockObj)
+            {
+                documentedAssemblies.Add(assemblyName);
+            }
+        }
+
+        /// <summary>
+        /// Attempts to cache the contents of a file formatted in the style of the standard .xml documentation file.
+        /// </summary>
+        /// <param name="filePath">The file location.</param>
+        /// <returns>Returns true if the file was successfully cached or empty.</returns>
+        private bool CacheXmlDocumentation(string filePath)
+        {
+            if (!File.Exists(filePath))
+            {
+                return false;
+            }
+
+            try
+            {
+                using (var reader = new StreamReader(filePath))
                 {
-                    using (var reader = new StreamReader(docFile))
+                    var doc = new XmlDocument();
+                    doc.Load(reader);
+
+                    foreach (XmlNode node in doc.GetElementsByTagName("userdoc"))
                     {
-                        while (!reader.EndOfStream)
+                        var key = node.ParentNode.Attributes["name"]?.Value;
+                        var documentation = node.InnerText.Trim();
+
+                        if (string.IsNullOrWhiteSpace(key) || string.IsNullOrWhiteSpace(documentation))
                         {
-                            var line = reader.ReadLine();
-                            if (string.IsNullOrWhiteSpace(line))
-                                continue;
-
-                            var separator = line.IndexOf('=');
-                            // TODO: Emit a warning here.
-                            if (separator < 0 || separator >= line.Length - 1)
-                                continue;
-
-                            var key = line.Substring(0, separator);
-                            var documentation = line.Substring(separator + 1);
-                            cachedDocumentations[key] = documentation;
+                            continue;
                         }
+
+                        cachedDocumentations.TryAdd(key, documentation);
```

Comment:
``cachedDocumentations[key] = documentation;`` otherwise documentation changes will keep the initial value